msm/README.md

```diff
@@ -49,3 +49,32 @@ x_3 & = \lambda^2 - x_{1} - x_{2} \\
 y_3 & = \lambda (x_{1} - x_{3}) - y_{1}
 \end{align}
 ```
+
+
+### Step 1 - Prove the decomposition in $l$ limbs
+
+The first circuit will prove the decomposition given by the foreign field gate
+implemented in Kimchi into the limbs required by the MSM circuit.
+The current implementation of the foreign field gate in Kimchi works with limbs on 88 bits.
+Therefore, we need to make a circuit that, for each bulletproof challenges,
+compute the decomposition in the new base. We will use one row = one decomposition.
+
+We have the following circuit shape:
+
+| $b_{1}$ | $b_{2}$ | $b_{3}$ | $c_{1}$ | $c_{2}$ | ... | $c_{17}$ |
+| ------- | ------- | ------- | ------- | ------- | --- | -------- |
+| ...     | ...     | ...     | ...     | ...     | ... | ...      |
+| ...     | ...     | ...     | ...     | ...     | ... | ...      |
+
+The constraints for each row should be something like:
+
+$$\sum_{i = 1}^3 b_{i} - \sum_{j = 1}^{17} c_{j} = 0$$
```

Comment:
1. This needs to be multiplied by corresponding `2^{88}` or `2^{16}`?
2. The circuit is a bit more complex if I understand your intention. I imagine it as follows: we have $3 * 15$ rows, so that each row contains all the input coefficients $b_i$ in three limbs ${b_{i,j}}_{j=1}^3$. Each row recombines them. The coefficient on line $i$ is then $c_i$ (in 17 limbs), and it's equal to $c_i = \prod_j b_j^{j_i}` where `j_i` is the `i`th bit of `j`. The prod is FF arithmetics. So we need to first as you said convert the `b` to 17 limbs. How does this sound?

Reply:
1. Yes. Updating

Reply:
2. yes, it should be more complicated, I'm in favor of using more columns than more rows.

Reply:
We can decompose $b_{2}$ in 22 chunks of 4 bits:
- $b_{2} = b_{2, 0} + b_{2, 1} 2^4 + ... + b_{2, 21} 2^{4 * 21}$
- Check that $b_{2, i}$ is between 0 and 15

And we constrain:
1. $c_{0} + c_{1} 2^{15} + ... + c_{11} 2^{15 * 11} = b_{0} + b_{1} 2^{88} + b_{2, 0} 2^{88 * 2}$
2. $c_{12} + c_{13} 2^{15} + ... + c_{17} 2^{15 * 6} = b_{2, 1} + b_{2, 2} * 2^4 + ... + b_{2, 21} * 2^{4 * 20}$

We could even simplify by assuming that $b_{2}$ is only restricted to values 79 bits.

Reply:
For the polynomials, see https://github.com/o1-labs/proof-systems/pull/1840. This PR is part of https://github.com/MinaProtocol/mina/issues/15214